Backend/src/main/java/com/example/backend/project/controller/ProjectController.java

```diff
@@ -43,6 +45,22 @@ public ResponseEntity getAllProjects(@RequestHeader String googleTokenEncoded){
         return new ResponseEntity(projectService.findAll(), HttpStatus.OK);
     }
 
+    @PutMapping("/softLock/{projectId}")
+    public ResponseEntity<Boolean> softLockProject(@RequestHeader String googleTokenEncoded, @PathVariable Long projectId){
+        List<AUTHORITY> authorities = List.of(AUTHORITY.MODERATOR, AUTHORITY.ADMINISTRATOR);
+        String email = JwtVerifier.verifyAndReturnEmail(googleTokenEncoded);
+        if (email == null)
+            return new ResponseEntity("Token is missing or invalid", HttpStatus.UNAUTHORIZED);
+        AppUser user = userService.findByEmail(email);
+        if (user == null)
+            return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
+        if (!authorities.contains(user.getAuthority()))
+            return new ResponseEntity("You don't have premission to this resource", HttpStatus.UNAUTHORIZED);
+
+        if (!projectService.existsById(projectId)) return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
+        return new ResponseEntity<>(projectService.softLockProject(projectId), HttpStatus.OK);
+    }
+
```

Comment:
nakon što mergam code refactoring granu na develop ovo će isto trebat refactorirat